docs/reference/contributing/guidelines/ci.md

```diff
@@ -0,0 +1,55 @@
+## Continuous integration
+
+Continuous integration (CI) means mainly automatic testing for pull requests
+
+### Travis CI
+
+In Mbed OS [Travis CI](https://travis-ci.org/ARMmbed/mbed-os) is used as primary automatic testing and checking run environment.
+
+Travis configuration is located in the [.travis.yml](https://github.com/ARMmbed/mbed-os/blob/master/.travis.yml) file in Mbed OS root directory. Mbed OS uses public travis so test results are publicly available and there are public [documentation available](https://docs.travis-ci.com/).
+
+#### Tests
+
+* **continuous-integration/travis-ci/pr** - Main run main
+* **travis-ci/astyle** - Check code style using [astyle](http://astyle.sourceforge.net/)
+* **travis-ci/docs** - [Doxygen](http://www.doxygen.org/) and naming checks:
+** Assert that the Doxygen build produced no warnings
+** Assert that all binary libraries are named correctly
+** Assert that all assebler files are named correctly
+* **travis-ci/events** - Check that Mbed OS compiles and run events tests
+* **travis-ci/gitattributestest** - Check that no changes after clone. This check that .gitattributes is used right way
+* **travis-ci/licence_check** - Checking that there is no GPL licence text in code
+* **travis-ci/littlefs** - Testing littlefs without embedded hardware
+* **travis-ci/tools-py2.7** - Run python tools tests with python 2.7
+
+### Jenkins
+
+[Jenkins](https://jenkins.io/) is used as internal testing and checking environment. Tests which have special requirements for execution enviroment are executed in internal Jenkins. In most cases test logs are published.
+
+#### How it works:
+* Jenkins uses a [scripted pipeline syntax](https://jenkins.io/doc/book/pipeline/)
+* Jenkins scripts are not publicly available. There is Jenkinsfile in Mbed OS root folder but that is just trigger for tests.
+* Jenkins selects required tests dynamically based on the code changes. For example: no tests are executed if only markdown (.md) file are changed
+* Jenkins run first small amount of tests to provide fast feedback and then additional tests
+
+#### Tests
+
+* **continuous-integration/jenkins/pr-head** Jenkins main pipeline script execution status
+* **jenkins-ci/build-ARM** - Build Mbed OS and examples with ARM. Related commands:
```

Comment:
@ARMmbed/mbed-os-maintainers What do you think about these checks inthe PR ?

My vote would be as it  is now (build, exporters, mbed 2 build, test) rather than each toolchain own one (would end up with 20 checks?)

Reply:
I don't see a particular problem with having up to 20 merge checks.
Plus, I think it would complicate the pipeline a bit to have the subjobs merge reporting after the fact.

Reply:
@cmonr reporting is done already.. sub jobs report the status on checks and main pipeline send the summary post in the end of run.